### ISSUE TITLE
Local server listens on all ip addresses

### ISSUE THREAD

**Issue by kenahrens** [GoogleCloudPlatform/functions-framework-nodejs#104]
On one of my dev setups I use Windows as my UI, but test code on a Linux machine on my local network. In my case I needed something that works over the network.

### Expected Behavior
Based on the README and what is printed out at runtime, I assumed it was binding only to `localhost` or the local loopback address.

### Actual Behavior
It turns out that the default behavior in Node is to listen on the unspecified IPv6 / IPv4 address (aka 0.0.0.0 for IPv4). Here's the docs from node:
https://nodejs.org/docs/latest-v10.x/api/net.html#net_server_listen_port_host_backlog_callback

Here is my quick test (against `nodejs-docs-samples` sample function):
```
$ npx @google-cloud/functions-framework --target=helloGET
Serving function...
Function: helloGET
URL: http://localhost:8080/
```

In another console on my Linux machine:
```
$ sudo netstat -anp | grep 8080
tcp6       0      0 :::8080                 :::*                    LISTEN      9041/node
$ ps -ef | grep 9041
kahrens   9041  3071  0 12:32 pts/1    00:00:00 node /home/kahrens/.nvm/versions/node/v10.16.0/bin/npx @google-cloud/functions-framework --target=helloGET
```

Because of this I sent in a PR to change the output of the console log to reflect the fact that it's listening on all IP addresses, not just localhost.
https://github.com/GoogleCloudPlatform/functions-framework-nodejs/pull/103

### Another Test With Express
And just to double check what is the expected behavior, I did a similar test with the Express hello world example, just the code from here: https://expressjs.com/en/starter/hello-world.html
```
$ node index.js
Example app listening on port 3000!
```
And then in another console:
```
$ sudo netstat -anp | grep 3000
tcp6       0      0 :::3000                 :::*                    LISTEN      10204/node
$ ps -ef | grep 10204
kahrens  10204  3071  0 12:37 pts/1    00:00:00 node index.js
```

**Comment by grant** [comment]
I don't understand what you mean by all IP addresses.
I am not extremely familiar with Windows networking, but I believe that 0.0.0.0 on Windows will listen to localhost/127.0.0.1.

I don't see the issue with multiple hostnames and I see no issue with logging `URL: http://localhost:8080/`. From my knowledge, it hasn't confused anyone else.

**Comment by kenahrens** [comment]
Hi @grant let me try to clarify.

If the app is bound to `localhost` or `127.0.0.1` and I'm running it on my Linux machine, then I can't access it from any other machine on the network. The project docs and server output seem to indicate the app is (only) binding to localhost which I found misleading. I was actually looking for some kind of `--flag` to set to tell it bind to all IPs, but I couldn't find it.

This is the part of the [NodeJS docs](https://nodejs.org/docs/latest-v10.x/api/net.html#net_server_listen_port_host_backlog_callback)I'm referring to specifically:

> If host is omitted, the server will accept connections on the unspecified IPv6 address (::) when IPv6 is available, or the unspecified IPv4 address (0.0.0.0) otherwise.

You can see [here](https://github.com/GoogleCloudPlatform/functions-framework-nodejs/blob/master/src/index.ts#L100) that the `host` value is omitted. So it's not listening on localhost, it's depending on the NodeJS default which is all IPs.

IMO printing `localhost` is incorrect, you can see from the output of `netstat` that it is listening on all IPs available, that's what `:::8080` (or `0.0.0.0:8080` for IPv4) means. If it was only only localhost it would have said `127.0.0.1:8080`.

**Comment by grant** [comment]
> then I can't access it from any other machine on the network

That is expected.

> app is (only) binding to localhost which I found misleading

That is common and purposeful. Why is that misleading?

I don't understand what the issue is with the Functions Framework.
Please file an issue talking about `localhost` with Express first and as this module uses `Express`. Those docs use the same terminology and our docs are very similar to theirs since it's a dependency.

https://github.com/expressjs/express
